luigi/contrib/pyspark_runner.py

```diff
@@ -51,14 +106,32 @@ def __init__(self, job, *args):
         self.args = args
 
     def run(self):
-        from pyspark import SparkContext, SparkConf
+        from pyspark import SparkConf
         conf = SparkConf()
         self.job.setup(conf)
-        with SparkContext(conf=conf) as sc:
+        with self._entry_point_class(conf) as (entry_point, sc):
             self.job.setup_remote(sc)
-            self.job.main(sc, *self.args)
+            self.job.main(entry_point, *self.args)
```

Comment:
So when using `use_spark_session`, the entrypoint here is a SparkSession object rather than a SparkContext. How can this be documented clearly to users? Or is it even necessary? Will `job.main()` just know how to handle its first arg input based on its Spark version?

Reply:
@dlstadther 
Short answer is that client code supposed to be aware of spark version it uses.
Long answer is:

when we want to use spark1 syntax we remain everything without any changes
when we decide to move onto  spark2+ we rewrite all our `PySparkTask` inheritors from `SparkContext` usage to `SparkSession` usage
e.g.

```
class X(PySparkTask):
    def main(self, sc, *args):
        sc.textFile('...').saveAsTextFile('...')
```

onto


```
class X(PySparkTask):
    def main(self, spark, *args):
        spark.read.text('...').write ....
```

and set `pyspark_runner.use_spark_session`option to True

Reply:
Ah i see. So there's more to be done than just adding the config setting. SparkSession and SparkContext do not provide identical signatures.